task/src/main/java/com/flipkart/flux/impl/task/LocalJvmTask.java

```diff
@@ -72,19 +69,28 @@ public int getExecutionTimeout() {
             While this works for methods with all unique param types, it
             will fail for methods where we have mutliple params of the same type.
              */
+            ClassLoader classLoader = toInvoke.getDeploymentUnitClassLoader() != null ? toInvoke.getDeploymentUnitClassLoader() : this.getClass().getClassLoader();
+            Class objectMapper = classLoader.loadClass("com.fasterxml.jackson.databind.ObjectMapper");
+
             for (int i = 0 ; i < parameterTypes.length ; i++) {
                 for (EventData anEvent : events) {
-                    if(Class.forName(anEvent.getType()).equals(parameterTypes[i])) {
-                        parameters[i] = objectMapper.readValue(anEvent.getData(), Class.forName(anEvent.getType()));
+                    if(Class.forName(anEvent.getType(), true, classLoader).equals(parameterTypes[i])) {
+                        parameters[i] = objectMapper.getMethod("readValue", String.class, Class.class).invoke(objectMapper.newInstance(), anEvent.getData(), Class.forName(anEvent.getType(), true, classLoader));
```

Comment:
we need to keep a single instance of object mapper available per class loader.
I have a solution in mind, but i want you to come up with one and we can compare :)